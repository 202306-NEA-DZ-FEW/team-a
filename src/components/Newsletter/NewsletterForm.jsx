import { useFormik } from "formik";
import { useRouter } from "next/router";
import { useTranslation } from "next-i18next";
import * as Yup from "yup";

import Input from "../Input";

function NewsletterForm() {
  const { t } = useTranslation();
  const formik = useFormik({
    initialValues: {
      email: "",
    },
    validationSchema: Yup.object({
      email: Yup.string().email("invalid email").required("email is required"),
    }),
    onSubmit: async (values) => {
      console.log("thank you", values);
    },
  });
  return (
    <div className='bg-neutral py-10 px-6  w-full'>
      <h3>Subscribe to our newsletter</h3>
      <form onSubmit={formik.handleSubmit}>
        <div>
          <Input
            name='email'
            type='email'
            placeholder='Enter your email'
            handleChange={formik.handleChange}
            value={formik.values.email}
            handleBlur={(e) => formik.dirty && formik.handleBlur(e)}
            error={formik.errors.email}
            touched={formik.touched.email}
          />
          {/* this is a spacer */}
          <div className='label' />
        </div>

        <button
          className='btn btn-primary self-center rounded-3xl text-white'
          type='submit'
          disabled={formik.isSubmitting}
        >
          {formik.isSubmitting ? "loading" : "Subscribe"}
        </button>
      </form>
    </div>
  );
}

export default NewsletterForm;
