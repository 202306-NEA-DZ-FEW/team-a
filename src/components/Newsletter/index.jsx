import NewsletterForm from "./NewsletterForm";

function Newsletter() {
  return <NewsletterForm />;
}

export default Newsletter;
