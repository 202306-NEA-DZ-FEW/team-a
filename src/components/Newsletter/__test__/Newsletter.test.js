import MailchimpSubscribe from "react-mailchimp-subscribe";
import renderer from "react-test-renderer";

import Newsletter from "..";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Newsletter>
        <MailchimpSubscribe url='' />
      </Newsletter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
