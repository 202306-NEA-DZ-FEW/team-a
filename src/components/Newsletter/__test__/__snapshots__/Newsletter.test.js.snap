// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-accent py-20 px-6 w-full"
  style={
    Object {
      "backgroundImage": "url(/img.jpg)",
      "backgroundSize": "cover",
    }
  }
>
  <h2
    className="text-3xl md:text-4xl tracking-wider font-extrabold text-white"
  >
    newsletterTitle
  </h2>
  <p
    className="my-4 max-w-2xl font-light text-xl text-gray-200"
  >
    newsletterDescription
    <br />
    newsletterJoin
  </p>
  <form
    className="flex flex-col md:flex-row gap-2"
    onSubmit={[Function]}
  >
    <div>
      <div>
        <div
          className="label ml-2"
        >
          <label
            className="label-text"
            htmlFor="email"
          />
          <label
            className="label-text-alt "
          />
        </div>
        <input
          className="input bg-white input-primary w-full rounded-full border-opacity-25"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="emailPlaceholder"
          type="email"
          value=""
        />
      </div>
      <div
        className="label"
      />
    </div>
    <button
      className="btn btn-neutral w-full md:w-fit self-center rounded-3xl"
      disabled={false}
      type="submit"
    >
      subscribe
    </button>
  </form>
</div>
`;
